liquidctl/driver/lianli_uni.py

```diff
@@ -0,0 +1,239 @@
+            raise ValueError(
+                f"channel must be between {_MIN_CHANNEL} and {_MAX_CHANNEL} (zero-based index)"
+            )
+
+        # Determine the desired action
+        if desired_state is None:
+            # Toggle the current state
+            desired_state = not self.pwm_channels[channel]
+
+        if desired_state:
+            debug_string = "Enabling"
+            channel_byte = 0x11 << channel  # enables PWM
+        else:
+            debug_string = "Disabling"
+            channel_byte = 0x10 << channel  # disables PWM
+
+        # Construct the command to toggle PWM synchronization
+        command_prefix = _PWM_COMMANDS.get(self.device_type)
+        if not command_prefix:
+            raise NotSupportedByDevice(
+                f"Unsupported device type for PWM sync: {self.device_type}"
+            )
+
+        command = command_prefix + [channel_byte]
+        _LOGGER.debug(
+            "%s PWM sync for channel %d: command %s", debug_string, channel, command
+        )
+
+        try:
+            self.device.write(command)
+            self.pwm_channels[channel] = desired_state  # Update state
+        except Exception as e:
+            _LOGGER.warning("Error writing to device: %s", e)
+
+    def set_fixed_speed(self, channel, duty, **kwargs):
+        """Set a fixed speed for the specified channel.
+
+        Parameters:
+            channel: str or int - The name of the channel (e.g., 'channel1') or the zero-based index of the channel
+            duty: int or float - The desired speed percentage (0-100)
+        """
+        if isinstance(channel, str):
+            channel_index = self._extract_channel_index(channel)
+        else:
+            channel_index = channel
+
+        if not _MIN_CHANNEL <= channel_index <= _MAX_CHANNEL:
+            raise ValueError(
+                f"channel must be between {_MIN_CHANNEL} and {_MAX_CHANNEL} (zero-based index)"
+            )
+
+        # Check if PWM is on for the channel
+        if self.pwm_channels[channel_index]:
+            _LOGGER.warning(
+                f"Cannot set fixed speed for Channel {channel_index + 1}: PWM is enabled. "
+                "Please disable PWM first to set a fixed speed."
+            )
+            return
+
+        duty = clamp(duty, 0, 100)
+        speed_byte = self._calculate_speed_byte(duty)
+        command = [224, channel_index + 32, 0, speed_byte]
+        _LOGGER.debug(
+            "Setting fixed speed for channel %d: duty %.1f%%, command %s",
+            channel_index,
+            duty,
+            command,
+        )
+        try:
+            self.device.write(command)
+        except Exception as e:
+            _LOGGER.error("Error writing to device: %s", e)
+        time.sleep(0.1)  # Delay to prevent race conditions
+
+        self.channel_speeds[channel_index] = duty
+
+        _LOGGER.info(f"Fan speed for Channel {channel_index + 1} set to {duty}%")
```

Comment:
It would be more consistent with other driver (if perhaps uglier):

```suggestion
       _LOGGER.info('setting %s PWM duty to %d%%', channel_index + 1, duty)
```

Also note that using f-strings in log entries is sub-optimal, because f-strings are unconditionally interpolated. Unfortunately, `logging` only really efficiently supports old style string interpolation.